Xamarin.PropertyEditing/Numeric.cs

```diff
@@ -40,110 +49,40 @@ private static void SetupMath ()
 				t = Nullable.GetUnderlyingType (t);
 			}
 
-			Task<Func<T, T>> createIncrement = Task.Run (() => {
-				var add = new DynamicMethod ("Add", t, new[] { t });
-				ILGenerator gen = add.GetILGenerator ();
-				gen.Emit (OpCodes.Ldarg_0);
-
-				switch (Type.GetTypeCode (t)) {
-				case TypeCode.Byte:
-				case TypeCode.UInt16:
-				case TypeCode.UInt32:
-					gen.Emit (OpCodes.Ldc_I4_1);
-					gen.Emit (OpCodes.Add_Ovf_Un);
-					break;
-				case TypeCode.UInt64:
-					gen.Emit (OpCodes.Ldc_I8, 1L);
-					gen.Emit (OpCodes.Add_Ovf_Un);
-					break;
-				case TypeCode.SByte:
-				case TypeCode.Int16:
-				case TypeCode.Int32:
-					gen.Emit (OpCodes.Ldc_I4_1);
-					gen.Emit (OpCodes.Add_Ovf);
-					break;
-				case TypeCode.Int64:
-					gen.Emit (OpCodes.Ldc_I8, 1L);
-					gen.Emit (OpCodes.Add_Ovf);
-					break;
-				case TypeCode.Single:
-					gen.Emit (OpCodes.Ldc_R4, 1f);
-					gen.Emit (OpCodes.Add);
-					break;
-				case TypeCode.Double:
-					gen.Emit (OpCodes.Ldc_R8, 1d);
-					gen.Emit (OpCodes.Add);
-					break;
-				default:
-					throw new NotSupportedException();
-				}
+			Task<Func<T,T>> createIncrement = Task.Run (() => {
+				var shiftby = Expression.Convert (Expression.Constant (1), t);
+				var value = Expression.Parameter (t);
 
-				gen.Emit (OpCodes.Ret);
-				if (!isNullable)
-					return (Func<T,T>)add.CreateDelegate (typeof(Func<T, T>));
-				else {
-					Delegate a = add.CreateDelegate (typeof(Func<,>).MakeGenericType (new[] { t, t }));
-					return (v) => {
-						if (v == null)
-							return (T)Activator.CreateInstance (t);
-						else {
-							return (T)a.DynamicInvoke (NullConverter.ConvertFrom (v));
-						}
-					};
+				Expression add;
```

Comment:
nit: it looks a bit weird not to use a ternary operator here (and for the sub expression below).